### PR TITLE
Reduce bundle size

### PR DESCRIPTION
**Pull request by kidonng**
Importing styles directly from `@primer/css` results in tons of unused CSS variables and rules being included in the bundle.

This PR tries to reduce the bundle size by patching the colors stylesheet to only include colors we actually use:

```diff
- dist/main.css    115.8 KB    1.21s
+ dist/main.css    14.2 KB    845ms
```

### CODE REVIEW

File: style/main.scss [hunk 1]
```diff
@@ -1,30 +1,13 @@
...
-  background-color: var(--color-canvas-default);
-  color: var(--color-fg-default);
-
-  ol {
-    list-style: decimal;
-  }
-  
-  ul {
-    list-style: disc;
-  }
```
Comment (kidonng) [inline style/main.scss:17]: This is added in f4c447091d1fc527cea475c06b1ad6579c987049 and actually not necessary.

I assume you looked at a website with CSS reset (e.g. [Tailwind's Preflight](https://tailwindcss.com/docs/preflight)) and made the wrong assumption that they are needed.

File: style/makefile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+patch:
+	mkdir -p dist
+	# Extract variables used by Markdown
+	grep -Eoh  -- "--[[:alnum:]-]+" node_modules/@primer/css/markdown/*.scss | sort -u > dist/variables
+	# Extract variables used by ourselves
+	grep -Eoh  -- "--[[:alnum:]-]+" main.scss | sort -u >> dist/variables
+	# Patch light colors
+	echo "@mixin primer-colors-light {\n  & {\n`grep -f dist/variables node_modules/@primer/primitives/dist/scss/colors/_light.scss`\n  }\n}" > node_modules/@primer/primitives/dist/scss/colors/_light.scss
+	# Patch dark colors
+	echo "@mixin primer-colors-dark {\n  & {\n`grep -f dist/variables node_modules/@primer/primitives/dist/scss/colors/_dark.scss`\n  }\n}" > node_modules/@primer/primitives/dist/scss/colors/_dark.scss
```
Comment (lucacasonato) [inline style/makefile:10]: Instead of this being a makefile, could you turn this into a Deno script?

**Review by lucacasonato — COMMENTED** [review]
Looks great!

File: style/makefile (hunk 2 above)
Comment (kidonng) [reply style/makefile]: Done

File: style/main.scss (hunk 1 above)
Comment (lucacasonato) [reply style/main.scss]: Why is this not needed? If you are using this on a site with Tailwind's preflight, won't the list style be unset?

Comment (kidonng) [reply style/main.scss]: https://github.com/denoland/dotland/pull/2040#issuecomment-1057151477

**Review by lucacasonato — APPROVED** [review]
LGTM, thanks a lot!

**Pull request merged**